host/main-window.hh

```diff
@@ -41,11 +42,20 @@ private:
    void togglePluginWindowVisibility();
    void recreatePluginWindow();
    void showAboutDialog();
+   void updateMenuItems();
 
    Application &_application;
    QWindow *_pluginViewWindow = nullptr;
    QWidget *_pluginViewWidget = nullptr;
 
+   QAction *_loadPluginPresetAction = nullptr;
+   QAction *_showPluginParametersAction = nullptr;
+   QAction *_showPluginQuickControlsAction = nullptr;
+   QAction *_togglePluginWindowVisibilityAction = nullptr;
+   QAction *_recreatePluginWindowAction = nullptr;
+
+   bool _pluginLoaded = false;
```

Comment:
I think the `_pluginLoaded` state + one of `updateMenuItems`/`onPluginLoadChange` could be eliminated:

only use one method, e.g. `setPluginSpecificMenuItemsEnabled(bool enabled)` (maybe you come up with a better name) and call it once for initialization and also connect it to a processor signal for the updates (see my comment below)

Reply:
Completely agree, I see now after using your suggested signal flow how the _pluginLoaded and onPluginLoadChange becomes obsolete.

I've opted to rename the method to the following:
updatePluginMenuItems(bool pluginLoaded)

1. It specifies these are plugin menu items
2. Function may do more than just enable/disable with pluginLoaded, so best to avoid having Enabled in the name

Fixed in latest commit

Reply:
By the way @Trinitou let me know if it's better you hit 'resolve conversation' rather than me :) Realise different projects prefer different approaches so however you prefer let me know :) cheers!

Reply:
> By the way @Trinitou let me know if it's better you hit 'resolve conversation' rather than me :) Realise different projects prefer different approaches so however you prefer let me know :) cheers!

I'm also just a contributor here btw :) Alex makes the rules here ;) . But I think it's totally fine (and IMO "more productive") if you resolve after resolving the issue.